data_model/src/query/mod.rs

```diff
@@ -165,6 +168,24 @@ pub mod model {
     )]
     #[ffi_type]
     pub struct MetadataValue(pub Value);
+
+    /// Request type clients (like http clients or wasm) can send to a query endpoint.
+    #[derive(Debug, Clone, Encode, Decode, Deserialize)]
+    pub enum QueryRequest<Q> {
```

Comment:
I believe this can also reside in `iroha_core` and be imported by cli

Reply:
Oh, I need to rename `WasmQueryRequest` to `SmartContractQueryRequest`

Reply:
I would annotate it with `transparent_api` and make `ClientQueryRequest` a newtype wrapper

Reply:
As we decidec in DM, it doesn't need to be under `transparent_api`, `#[model]` is enough.

I implemented wrappers